test/suite/test_prepare_lookaside02.py

```diff
@@ -0,0 +1,174 @@
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_prepare_lookaside02.py
+#   Prepare updates can be resolved for both commit / rollback operations.
+#
+
+from helper import copy_wiredtiger_home
+import random
+from suite_subprocess import suite_subprocess
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+def timestamp_str(t):
+    return '%x' % t
+
+class test_prepare_lookaside02(wttest.WiredTigerTestCase, suite_subprocess):
+    tablename = 'test_prepare_cursor'
+    uri = 'table:' + tablename
+    txn_config = 'isolation=snapshot'
+
+    types = [
+        ('col', dict(extra_config=',log=(enabled=false),key_format=r')),
+        ('row', dict(extra_config=',log=(enabled=false)')),
+        #('lsm', dict(extra_config=',log=(enabled=false),type=lsm')),
+    ]
+
+    # Transaction end types
+    txn_end = [
+        ('txn_commit', dict(txn_commit=True)),
+        ('txn_rollback', dict(txn_commit=False)),
+    ]
+
+    scenarios = make_scenarios(types, txn_end)
+    conn_config = 'log=(enabled)'
+
+    def test_prepare_conflict(self):
+        if not wiredtiger.timestamp_build():
+            self.skipTest('requires a timestamp build')
+
+        self.session.create(self.uri,
+            'key_format=i,value_format=i' + self.extra_config)
+        c = self.session.open_cursor(self.uri)
+
+        # Insert keys 1..100 each with timestamp=key, in some order
+        orig_keys = range(1, 101)
+        keys = orig_keys[:]
+        random.shuffle(keys)
+
+        # Scenario: 1
+        # Check insert operation
+        self.session.begin_transaction(self.txn_config)
+        c[1] = 1
+        # update the value with in this txn
+        self.session.prepare_transaction('prepare_timestamp=' + timestamp_str(100))
+        if self.txn_commit == True:
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(101))
+        else:
+            self.session.rollback_transaction()
+
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(150))
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(150))
+        self.session.checkpoint()
+
+        # Scenario: 2
+        # Check update operation
+        #   update a existing key.
+        #   update a newly inserted key with in this transaction
+        self.session.begin_transaction(self.txn_config)
+        # update a committed value, key 1 is inserted above.
+        c[1] = 2
+        # update a uncommitted value, insert and update a key.
+        c[2] = 1
+        c[2] = 2
+        self.session.prepare_transaction('prepare_timestamp=' + timestamp_str(200))
+        if self.txn_commit == True:
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(201))
+        else:
+            self.session.rollback_transaction()
+
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(250))
```

Comment:
same as before, if we don't need to, lets avoid setting timestamp and checkpoints.